neps/nep-0568.md

```diff
@@ -135,6 +135,13 @@ supporting smooth transitions without altering storage structures directly.
 
 ### State Sync
 
+Changes to the state sync protocol aren't typically conisdered protocol changes requiring a version bump, since it's concerned with downloading state that isn't present locally, rather than with the rules of execution of blocks and chunks. But it might still be helpful to outline some planned changes to state sync intended to make the resharding implementation easier to work with.
+
+When nodes sync state (either because they've fallen far behind the chain, or because they're going to become a chunk producer for a new shard in a future epoch), they first identify a point in the chain they'd like to sync to. This is always the first block of the current epoch, which the node should be aware of once it has synced headers to the current point in the chain. The hash of this first block is referred to as the "sync_hash" in many places in the state sync implementation. Then the node makes a request (currently to centralized storage on GCS, but in the future to other nodes in the network) for a `ShardStateSyncResponseHeader` corresponding to that "sync_hash" and
+the Shard ID of the shard it's interested in. Among other things, this header includes the last new chunk before "sync_hash" in the shard, and a `StateRootNode` with hash equal to that chunk's `prev_state_root` field. Then the node downloads (again from GCS, but in the future it'll be from other nodes) the nodes of the trie with that `StateRootNode` as its root. Afterwards, it applies new chunks in the shard until it's caught up.
```

Comment:
This section may be a bit too technical for the specification. Perhaps you can move the implementation details to the "reference implementation" section below and here keep it at a higher level?

Reply:
yea good point, done